### ISSUE TITLE
Accessibility of the Spawn_pending page

### ISSUE THREAD

**Issue by bl-aire** [jupyterhub/jupyterhub#4279]
### Details from report:
Based on evaluation using the WAVE accessibility tool, the following are responsible for accessibility issues on the Spawn_pending page:

- **Reoccurring accessibility issues**: These include issues like a very **Low contrast** between nav items and their background, **Language missing or invalid** and **Missing first level header** which have been fixed in PR #4274

- **Skipped heading level**: A modal's title on the page has a `h4` element. So the current hierarchy is `<h1> ----> <h4>`

### Report:

<img width="1280" alt="Screenshot 2022-12-20 at 12 53 04" src="https://user-images.githubusercontent.com/84882370/208909700-41a86293-e3ad-4839-b8aa-ead83796ef87.png">

@minrk

**Comment by minrk** [comment]
I think the right fix here is to remove the `h4` for the error. [This answer](https://ux.stackexchange.com/questions/105439/heading-levels-in-an-accessible-modal-window) suggests that modals should use an h1, even though that means multiple h1s on the page. But it needs to be clear to screen readers that this content is not visible (It's under `aria-hidden="true"`, I'm not sure if it's enough).

Either that, or we should change how we do modals so that the HTML is simply _not present_ until the modal, rather than hidden.

- Modal dialog HTML is defined here: https://github.com/jupyterhub/jupyterhub/blob/eca9c00872cbf5faa2f9b81b1ff9f5f30f610554/share/jupyterhub/templates/page.html#L4-L20
- added to the page here: https://github.com/jupyterhub/jupyterhub/blob/eca9c00872cbf5faa2f9b81b1ff9f5f30f610554/share/jupyterhub/templates/page.html#L183-L187
- made visible us from js here: https://github.com/jupyterhub/jupyterhub/blob/eca9c00872cbf5faa2f9b81b1ff9f5f30f610554/share/jupyterhub/static/js/utils.js#L119-L121

It is in a macro because it was used several times in the old admin page which was a very complex template, but we no longer have that, so it's used only once now.

Maybe the way to do it is to construct the whole HTML dialog in javascript instead (i.e. replace the modal jinja macro with a modal _function_ in utils.js).

**Comment by bl-aire** [comment]
> I think the right fix here is to remove the `h4` for the error. [This answer](https://ux.stackexchange.com/questions/105439/heading-levels-in-an-accessible-modal-window) suggests that modals should use an h1, even though that means multiple h1s on the page. But it needs to be clear to screen readers that this content is not visible (It's under `aria-hidden="true"`, I'm not sure if it's enough).

Personally, I think this works because I have accessed a couple of pages with multiple `h1`'s that weren't flagged as an accessibility issue by WAVE. 

> Either that, or we should change how we do modals so that the HTML is simply _not present_ until the modal, rather than hidden.

I can research which approach is best or ask for guidance from members of the accessibility team. They have been very helpful.
